tests/test_solver.py

```
@@ -212,56 +218,98 @@ def test_circular_conflict_edge():
+def test_solver_soft_groupers_clamping():
+    """Trigger the cohesion penalty clamping logic with a safe but high weight."""
+    p = make_participants(4, groupers=["T", "T", "", ""])
     cfg = SolverConfig(
-        num_groups=2, group_capacities=[1, 1], score_weights={"S1": 1.0, "S2": 1.0}
+        num_groups=2,
+        group_capacities=[2, 2],
+        score_weights={SCORE_COL: 1.0},
+        grouper_weight=1_000,
     )
-    from src.core.solver import AdvancedScoring, ConstraintBuilder
+    results, status, _ = solver.solve_with_ortools(p, cfg)
```

Ruff (0.15.12)

[warning] Unpacked variable `results` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)